plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueTableOperations.java

```
@@ -0,0 +1,236 @@
+                .withParameters(parameters);
+
+        boolean succeeded = false;
+        try {
+            Table table = getTable();
+
+            checkState(currentMetadataLocation != null, "No current metadata location for existing table");
+            String metadataLocation = table.getParameters().get(METADATA_LOCATION);
+            if (!currentMetadataLocation.equals(metadataLocation)) {
+                throw new CommitFailedException("Metadata location [%s] is not same as table metadata location [%s] for %s",
+                        currentMetadataLocation, metadataLocation, getSchemaTableName());
+            }
+
+            TableInput tableInputToUpdate = tableInput
+                    .withDescription(table.getDescription())
+                    .withTargetTable(table.getTargetTable())
+                    .withLastAccessTime(table.getLastAccessTime())
+                    .withLastAnalyzedTime(table.getLastAnalyzedTime())
+                    .withPartitionKeys(table.getPartitionKeys())
+                    .withRetention(table.getRetention())
+                    .withStorageDescriptor(table.getStorageDescriptor())
+                    .withViewExpandedText(table.getViewExpandedText())
+                    .withViewOriginalText(table.getViewOriginalText());
+
+            stats.getUpdateTable().call(() -> {
+                glueClient.updateTable(new UpdateTableRequest()
+                        .withCatalogId(catalogId)
+                        .withDatabaseName(database)
+                        .withTableInput(tableInputToUpdate));
+                return null;
+            });
+            succeeded = true;
+        }
+        catch (ConcurrentModificationException | CommitFailedException e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, format("Cannot commit %s because of concurrent update", getSchemaTableName()), e);
+        }
+        finally {
+            cleanupMetadataLocation(!succeeded, newMetadataLocation);
```

this can swallow exception in flight.